src/internal/ppsutil/decoder.go

```diff
@@ -16,23 +18,78 @@ import (
 type PipelineManifestReader struct {
 	// Do first round of parsing (yaml -> holder) here (not serde), in case of
 	// multi-pipeline document
-	decoder *yaml.Decoder
-	next    func() (*ppsclient.CreatePipelineRequest, error)
+	decoder    *yaml.Decoder
+	next       func() (*ppsclient.CreatePipelineRequest, error)
+	noValidate bool
 }
 
-// NewPipelineManifestReader creates a new manifest reader from a path.
-func NewPipelineManifestReader(pipelineBytes []byte) (result *PipelineManifestReader, retErr error) {
+// NewPipelineManifestReader creates a new manifest reader which reads manifests
+// from an io.Reader.
+func NewPipelineManifestReader(r io.Reader) (result *PipelineManifestReader, retErr error) {
 	return &PipelineManifestReader{
-		decoder: yaml.NewDecoder(bytes.NewReader(pipelineBytes)),
+		decoder: yaml.NewDecoder(r),
 	}, nil
 }
 
-func convertRequest(request interface{}) (*ppsclient.CreatePipelineRequest, error) {
-	var result ppsclient.CreatePipelineRequest
+// DisableValidation disables pipeline validation.
+//
+// TODO(INT-1006): this exists only because the implementation of the /_mount_datums
+// endpoint in the FUSE server parses its PUT body as a full pipeline spec.
+func (r *PipelineManifestReader) DisableValidation() *PipelineManifestReader {
+	r.noValidate = true
+	return r
+}
+
+type unvalidatedCreatePipelineRequest ppsclient.CreatePipelineRequest
```

Comment:
I haven't tried it, but you might be able to get rid of `Reset()`, `String()`, and `ProtoMessage()` below by declaring the type like:
```
type unvalidatedCreatePipelineRequest struct {
  *ppsclient.CreatePipelineRequest
}
```
(and then use raw `unvalidatedCreatePipelineRequest` throughout to avoid double-dereferencing).

I think the in-memory representation is the same, but it does look a little weirder; the wrapper struct just exists to pass the methods through. Your call if it seems better or worse; just wanted to suggest it.

Reply:
Unfortunately, that appears to interact badly with `serde.RoundTrip`; preventing it from realizing that it’s a `proto.Message` when unmarshalling. I _could_ unmarshal the data as a `ppsclient.CreatePipelineRequest` and wrap it if it needs to be validated, or just get rid of `unvalidatedCreatePipelineRequest`.

This was an attempt to use type safety to prevent unvalidated data escaping the decoder, but perhaps that is a bit too “clever”?

Reply:
Ah, I mean framed that way, it does make sense. On reflection, it also makes sense (as a Go feature) that you wouldn’t necessarily want to pass the wrapped type’s methods through if the wrapper type carries safety properties. LGTM